test/new-e2e/tests/npm/ec2_1host_test.go

```diff
@@ -0,0 +1,160 @@
+	e2e.Run(t, s, e2e.FakeIntakeStackDef(nil, agentparams.WithSystemProbeConfig(NPMsystemProbeConfig)), e2eParams...)
+}
+
+// TestFakeIntakeNPM Validate the agent can communicate with the (fake) backend and send connections every 30 seconds
+//   - looking for 1 host to send CollectorConnections payload to the fakeintake
+//   - looking for 3 payloads and check if the last 2 have a span of 30s +/- 500ms
+func (v *ec2VMSuite) TestFakeIntakeNPM() {
+	t := v.T()
+
+	// default is to reset the current state of the fakeintake aggregators
+	if !v.DevMode {
+		v.Env().Fakeintake.FlushServerAndResetAggregators()
+	}
+
+	targetHostnameNetID := ""
+	// looking for 1 host to send CollectorConnections payload to the fakeintake
+	v.EventuallyWithT(func(c *assert.CollectT) {
+		// generate a connection
+		v.Env().VM.Execute("curl http://www.datadoghq.com")
+
+		hostnameNetID, err := v.Env().Fakeintake.GetConnectionsNames()
+		assert.NoError(c, err, "GetConnectionsNames() errors")
+		assert.NotZero(c, len(hostnameNetID), "no connections yet")
+		targetHostnameNetID = hostnameNetID[0]
+
+		t.Logf("hostname+networkID %v seen connections", hostnameNetID)
+	}, 60*time.Second, time.Second, "no connections received")
+
+	// looking for 3 payloads and check if the last 2 have a span of 30s +/- 500ms
+	v.EventuallyWithT(func(c *assert.CollectT) {
+		cnx, err := v.Env().Fakeintake.GetConnections()
+		assert.NoError(t, err)
+
+		require.Greater(c, len(cnx.GetPayloadsByName(targetHostnameNetID)), 2, "not enough payloads")
+		var payloadsTimestamps []time.Time
+		for _, cc := range cnx.GetPayloadsByName(targetHostnameNetID) {
+			payloadsTimestamps = append(payloadsTimestamps, cc.GetCollectedTime())
+		}
+		dt := float64(payloadsTimestamps[2].Sub(payloadsTimestamps[1])) / float64(time.Second)
+		t.Logf("hostname+networkID %v diff time %f seconds", targetHostnameNetID, dt)
+
+		// we want the test fail now, not retrying on the next payloads
+		require.Greater(t, math.Abs(dt-30), 0.5, "delta between collection is higher that 500ms")
+	}, 90*time.Second, time.Second, "no connections received")
+}
+
+// TestFakeIntakeNPM_TCP_UDP_DNS validate we received tcp, udp, and DNS connections
+// with some basic checks, like IPs/Ports present, DNS query has been captured, ...
+func (v *ec2VMSuite) TestFakeIntakeNPM_TCP_UDP_DNS() {
+	t := v.T()
+
+	v.EventuallyWithT(func(c *assert.CollectT) {
+		// generate connections
+		v.Env().VM.Execute("curl http://www.datadoghq.com")
+		v.Env().VM.Execute("dig @8.8.8.8 www.google.ch")
+
+		cnx, err := v.Env().Fakeintake.GetConnections()
+		assert.NoError(c, err)
+
+		var currentHostname string
+		var currentConnection *agentmodel.Connection
+		t.Cleanup(func() {
+			if t.Failed() {
+				t.Logf(krpretty.Sprintf("test failed on host %s at connection %# v", currentHostname, currentConnection))
+			}
+		})
+
+		foundDNS := false
+		cnx.ForeachConnection(func(c *agentmodel.Connection, cc *agentmodel.CollectorConnections, hostname string) {
```

Comment:
Question: is this looping through all connections that the fakeIntake might have in memory? Potentially from other tests? 

It feels wrong that we need to use ForEachConnection to figure out what the hostname running the agent is. Could we instead ask the FakeIntake what the connections for the agent host *only* are ?

Reply:
fakeintake data would be cleanup when the test start via `v.Env().Fakeintake.FlushServerAndResetAggregators()`
https://github.com/DataDog/datadog-agent/pull/19449/files#diff-932c1db6963f2ddda0a1ad276347d9dce7630f2c6ca763338b8651687fdf945dR104

Reply:
I think this would read a lot cleaner if at line 112 we could write:

```
cnx, err := v.Env().Fakeintake.GetConnectionsForHost(v.Env().VM.Name)
```

Then we wouldn't need the bookkeeping introduced in `countConnections := make(map[string]*countCnx)`

Reply:
Note you can filter requests to the fakeintake, out of the box by tag and name, you can add custom filter to the optional args

Reply:
VM.Name (aws ec2 name like aws-ami-0fc5d935ebf8bc3bc)  is not really the name that will match, for the moment it hostname and networkid reported by the agent
https://github.com/DataDog/datadog-agent/pull/19449/files#diff-e1397ba5c0f000fa4d869b6832deebf012d199d6f622b5121a9a54a5409b0a85R25

like "i-0100f099360d3a374/vpc-029c0faf8f49dee8d"

Reply:
> Note you can filter requests to the fakeintake, out of the box by tag and name, you can add custom filter to the optional args

Yes I seen the generic and thanks for that, but I don't know the agent hostname or network representation, so I can't filter on a unknown key